src/storage/sql_table.cpp

```diff
@@ -234,8 +236,9 @@ std::pair<bool, storage::TupleSlot> SqlTable::Update(transaction::TransactionCon
     auto old_pair = InitializerForProjectedRow(old_col_oids, old_version);
     auto old_buffer = common::AllocationUtil::AllocateAligned(old_pair.first.ProjectedRowSize());
     ProjectedRow *old_pr = old_pair.first.InitializeRow(old_buffer);
-    bool valid = Select(txn, slot, old_pr, old_pair.second, old_version);
```

Comment:
Why not use the SqlTable select. This way you don't have to reason about the old version's projected row. The select function will handle all of that and give you back the data in a projected row of the latest version.

Reply:
- I still need to reason about the old version's projected row for SqlTable select because I need to know the version number and allocate appropriate buffers. We know which data table to look into at this point so I just go directly to that data table, without going through SqlTable select. 
- It keeps a clear invariant that "the version number passed into SqlTable operations should match the version which the transaction can see". This is also related to the Scan. We can talk about it offline. 
- Select doesn't reason about default values. If you are talking about inserts, in theory, each DataTableVersion has a map for default values so DataTable has the ability to fill out default values.